### PR TITLE
coverage: use source-based coverage

### PR DESCRIPTION
**Pull request by davidhewitt**
The [source-based coverage](https://doc.rust-lang.org/nightly/unstable-book/compiler-flags/source-based-code-coverage.html) in rust nightly is meant to be better than the "current" `-Zprofile` option, so let's try it here!

### CODE REVIEW

**Comment by davidhewitt** [comment]
Well, looks like it'll make a tiny positive improvement to our coverage numbers! 😄

**Comment by davidhewitt** [comment]
In addition I guess it means we don't need to support running our test suite with `panic = "abort"` any more, so I pushed a separate commit to remove that...

File: src/err/mod.rs [hunk 1]
```diff
@@ -557,12 +557,6 @@ mod tests {
     #[test]
     #[should_panic(expected = "new panic")]
     fn fetching_panic_exception_resumes_unwind() {
-        // TODO replace with #[cfg(panic = "unwind")] once stable
-        if !crate::cfg_panic_unwind() {
-            // panic to meet the expected abort in panic=abort :-/
-            panic!("new panic");
-        }
-
```
Comment (kngwyu) [inline src/err/mod.rs:565]: Well I'm quite happy to see this is removed

**Review by kngwyu — APPROVED** [review]
Thanks!

**Pull request merged**